align.go

```diff
@@ -0,0 +1,115 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package plot
+
+import (
+	"math"
+
+	"github.com/gonum/plot/vg"
+	"github.com/gonum/plot/vg/draw"
+)
+
+// Align returns a two-dimensional row-major array of Canvases which will
+// produce tiled plots with DataCanvases that are evenly sized and spaced.
+// The arguments to the function are a two-dimensional row-major array
+// of plots, a tile configuration, and the canvas to which the tiled
+// plots are to be drawn.
+func Align(plots [][]*Plot, t draw.Tiles, dc draw.Canvas) [][]draw.Canvas {
+	o := make([][]draw.Canvas, len(plots))
+
+	// Create the initial tiles.
+	for j := 0; j < t.Rows; j++ {
+		o[j] = make([]draw.Canvas, len(plots[j]))
+		for i := 0; i < t.Cols; i++ {
+			o[j][i] = t.At(dc, i, j)
+		}
+	}
+
+	type posNeg struct {
+		p, n float64
+	}
+	xSpacing := make([]posNeg, t.Cols)
+	ySpacing := make([]posNeg, t.Rows)
+
+	// Calculate the maximum spacing between data canvases
+	// for each row and column.
+	for j, row := range plots {
+		for i, p := range row {
+			c := o[j][i]
```

Comment:
I think it's reasonable to have a `nil` `plot.Plot` in here. Check this and skip if `nil`.